### ISSUE TITLE
Tappable/clickable semantic nodes should have reasonable fallback role

### ISSUE THREAD

**Issue by yjbanov** [flutter/flutter#145434]
Some framework widgets, such as [ExpansionTile](https://github.com/flutter/flutter/issues/143852) and `Tab` (via [InkWell](https://github.com/flutter/flutter/blob/5c54509ad90ae7d9242a7baa1a43e6eeed9eaa75/packages/flutter/lib/src/material/ink_well.dart#L1327)), and widgets authored by developers, would add a tap action to their respective semantics node, but would not pick a concrete role for the node, such as button, checkbox, radiobox, text field, combobox, link, etc.

On the web this translates to an element with a `click` event listener but without an ARIA `role` attribute. Some screen readers would look for the listener and say "double-tap to activate" (or whatever is appropriate for the form factor of the device). Some will not communicate that the node is interactive (https://github.com/flutter/flutter/issues/143852), prompting custom solutions such as [this](https://github.com/flutter/flutter/blob/4d5e675253f21e6405bc325e17cca5221c1d654b/packages/flutter_localizations/lib/src/l10n/generated_material_localizations.dart#L7642).

## Proposal

When a tap action is present the framework always assigns a role to the node. If the developer does not specify one, the framework fallback to the most reasonable default (e.g. "button"). For the cases where it's preferable to communicate the role using a custom hint (like in the `ExpansionTile` case) we introduce a special role for that, in which case the engine would know that the role is communicated using means other than the ARIA role.

Taking this proposal to extreme, perhaps we should introduce a formal concept of "role" in the semantics tree. Currently the role is described using various flags that the engine pattern-matches on to assign it a formal role internally. [Here's how the web engine does it](https://github.com/flutter/engine/blob/dce639a4c453bc5682a630918b7606567243059e/lib/web_ui/lib/src/engine/semantics/semantics.dart#L1577). A formal role enum could improve the following:

- Remove the guesswork from the engine.
- All engine implementations would be in agreement about what role a particular semantics node plays.
- Enable better validation/testing of nodes. For example, should a checkbox have the `paste` action?
- Allow the engine to pick the best fallback when there platform does not directly support a role. Let's say we want to have a special role for `Tab`. ARIA does not define a formal "tab" role. However, there might be a combination of ARIA attributes that approximate it quite well. Right now, there's no way to tell if something is a tab, and the web engine just falls back to `GenericRole` that's also used for paragraphs of static text (which also doesn't have a role, and is pattern-matched onto "leaf node with a label").

**Comment by yjbanov** [comment]
Another use-case for a formal concept of role is when merging framework-side semantics node it would provide an unambiguous signal when two nodes can and cannot be merged. Essentially, if two nodes each has a role, the nodes are not merged. For example, something that claims to be a "button" would not be merged with something that claims to be a "paragraph". Today, they might.
